src/db/tables/subscriber_email_preferences.ts

```diff
@@ -261,11 +261,14 @@ async function unsubscribeMonthlyMonitorReportForUnsubscribeToken(
   try {
     sub = await getEmailPreferenceForUnsubscribeToken(unsubscribeToken);
 
-    if (sub.id && !sub.monthly_monitor_report_free) {
+    if (
```

Comment:
I think the previous code didn't take into account that `sub.monthly_monitor_report_free` could be `null` or `undefined`, hence these changes.